### ISSUE TITLE
Image version tag of "latest" dereferenced by ARG does not trigger DL3007

### ISSUE THREAD

**Issue by 4x0v7** [hadolint/hadolint#412]
- [x] This is a bug report
- [ ] This is a feature request
- [x] I searched existing issues before opening this one

### Expected behavior

DL3007 is triggered

### Actual behavior

DL3007 is not triggered

### Steps to reproduce the behavior

Have tested with both `brew` installed Hadolint and Docker using `docker run --rm -i hadolint/hadolint < Dockerfile`

**Output of `hadolint --version` or
  `docker run --rm hadolint/hadolint hadolint --version`:**

```bash
Haskell Dockerfile Linter 1.17.4-no-git (Brew)
Haskell Dockerfile Linter v1.17.5-8-gc8bf307 (Docker)
```

**Dockerfile** (if relevant)

```Dockerfile
ARG RUBY_VERSION=latest
FROM ruby:${RUBY_VERSION}
COPY Gemfile .
COPY Gemfile.lock .
RUN gem update bundler && \
    bundle install
```

**Additional environment details (OS, stack version, etc.)**

**Comment by 4x0v7** [comment]
I would guess you would have to first check that the part after the `:` is a variable, then check if that variable is already defined, then go check that it matches the string `latest`.
I am really interested in how this would be implemented in Haskell :)

**Comment by zemanlx** [comment]
Hi @4x0v7,

thank you for your interest in Haskell and hadolint. Unfortunately, hadolint is not processing variables or arguments right now so it does not know what is in that variable. What it does right now is a search for a tag and checks that the tag is not called `latest`. This space is much more complicated than it looks from the first sight, see [Environment replacement](https://docs.docker.com/engine/reference/builder/#environment-replacement) and also `ARG` can be replaced at build time and `ENV` at runtime. So even though hadolint checks it statically in Dockerfile it still can be changed later if it is not hardcoded.

However, if you feel like writing some Haskell, we would welcome contribution!

**Comment by 4x0v7** [comment]
Thank you @zemanlx for your reply!
Indeed I had not realised the number of cases you would potentially end up implementing to have this work for each valid Docker use-case.

I will close this issue in that case. Perhaps in the future if I learn some Haskell I have a challenge to solve :)
